src/conv.rs

```diff
@@ -268,6 +270,60 @@ unsafe impl<'a> IntoValue for &mut [u8] {
     }
 }
 
+macro_rules! array_impl {
+    ($n:tt) => {
+        unsafe impl FromValue<'static> for [u8; $n] {
+            fn from_value(value: Value) -> Self {
+                unsafe {
+                    let len = sys::caml_string_length(value.raw().0);
+                    assert!(len == $n);
+                    let ptr = sys::string_val(value.raw().0);
+                    ::core::slice::from_raw_parts(ptr, len).try_into().unwrap()
```

Comment:
Be careful here. If you create a slice by reusing the bytes from the OCaml heap, they may become invalid after the OCaml garbage collector runs (because the value may be moved around). If that happens and you something with this bytes, you will either get invalid data or crash the program.

Reply:
Hey! Thanks for the super quick review :) My understanding here is that the `try_into()` will copy all the content into a new array `[u8; n]` so there's no dangling reference. Am I missing something?

Reply:
@mimoo no, you are not missing anything, it was actually me being hasty misinterpreting what is going on here.